src/solvers/dg_curved/dg_2d.jl

```diff
@@ -25,24 +28,43 @@ function rhs!(du::AbstractArray{<:Any,4}, u, t,
 end
 
 
+# Return the contravariant vector in the specified `orientation` at the specified nodes and element,
+# multiplied by the Jacobian of the transformation mapping.
+function scaled_contravariant_vector(orientation, i, j, element, cache)
+  @unpack metric_terms = cache.elements
+
+  if orientation == 1
+    return SVector(metric_terms[2, 2, i, j, element], -metric_terms[1, 2, i, j, element])
+  end
+  
+  # orientation == 2
+  return SVector(-metric_terms[2, 1, i, j, element], metric_terms[1, 1, i, j, element])
+end
```

Comment:
Since this function is only used internally, would you mind changing its signature to
```julia
scaled_contravariant_vector(orientation, i, j, element, metric_terms)
```
My reasoning is as follows: We use `NamedTuple`s as `cache`. Thus, there will be many different types of `cache`s for different setups. Hence, multiple specializations of `scaled_contravariant_vector` need to be compiled if `cache` is an argument. Passing `cache.elements.metric_terms` should thus reduce the number of method specializations and thus also the latency a bit. This will probably  not have a huge impact, but we might try nevertheless.